packages/rum-core/test/common/truncate.spec.js

```diff
@@ -0,0 +1,313 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { truncate, truncateModel } from '../../src/common/truncate'
+
+/**
+ * Dummy models to make the testing easier
+ */
+const getMetadataModel = limit => ({
+  service: {
+    name: [limit, true],
+    version: [limit],
+    agent: {
+      version: [limit]
+    },
+    environment: [limit]
+  }
+})
+const getContextModel = limit => ({
+  user: {
+    id: [limit],
+    email: [limit],
+    username: [limit]
+  },
+  tags: {
+    '*': [limit]
+  }
+})
+const getSpanModel = limit => ({
+  name: [limit, true],
+  type: [limit, true],
+  subtype: [limit],
+  action: [limit],
+  id: [limit, true],
+  trace_id: [limit, true],
+  parent_id: [limit, true],
+  transaction_id: [limit, true],
+  context: getContextModel(limit)
+})
+
+const getTransactionModel = limit => ({
+  name: [limit],
+  parent_id: [limit],
+  type: [limit, true],
+  id: [limit, true],
+  trace_id: [limit, true],
+  span_count: {
+    started: [limit, true]
+  },
+  context: getContextModel(limit)
+})
+const getErrorModel = limit => ({
+  id: [limit, true],
+  trace_id: [limit],
+  transaction_id: [limit],
+  parent_id: [limit],
+  culprit: [limit],
+  exception: {
+    type: [limit]
+  },
+  transaction: {
+    type: [limit]
+  },
+  context: getContextModel(limit)
+})
+
+describe('Truncate', () => {
+  function generateStr(ch, length) {
+    return new Array(length + 1).join(ch)
+  }
+
+  it('truncate values with options', () => {
+    expect(truncate('', undefined, true)).toEqual('N/A')
+    expect(truncate(generateStr('ab', 5), 2)).toEqual('ab')
+    const placeHolder = 'dummyplaceholder'
+    expect(truncate('', undefined, true, placeHolder)).toEqual(placeHolder)
```

Comment:
Maybe some tests with undefined and null as well as if the field is not required!